nixos/lib/testing-python.nix

```diff
@@ -83,10 +83,9 @@ rec {
             "nixos-test-driver-${testName}";
 
       vlans = map (m: m.config.virtualisation.vlans) (lib.attrValues nodes);
-      vms = map (m: m.config.system.build.vm) (lib.attrValues nodes);
 
       nodeHostNames = let
-        nodesList = map (c: c.config.system.name) (lib.attrValues nodes);
+        nodesList = (lib.attrNames nodes);
       in nodesList ++ lib.optional (lib.length nodesList == 1 && !lib.elem "machine" nodesList) "machine";
 
       # TODO: This is an implementation error and needs fixing
```

Comment:
This comment was once added to acknowledge the problem that these names were too coupled.

When the `nodes.<name>` names correspond to python identifiers only, and to hostnames/system names only by their default value, we can get rid of this comment and `invalidNodeNames`.

We may still want to validate the attribute names to be valid identifiers, but as an alternative, we could use a dict of nodes in the test script. I don't have a much of an opinion about this choice.